MacOS builds are failing due to sccache

This seems to affect the `cargo clippy` step, with the only output being:

```
(… many other lines of "checking" …)
    Checking uu_printf v0.0.25 (/Users/runner/work/coreutils/coreutils/src/uu/printf)
    Checking uu_cat v0.0.25 (/Users/runner/work/coreutils/coreutils/src/uu/cat)
    Checking uu_expr v0.0.25 (/Users/runner/work/coreutils/coreutils/src/uu/expr)
    Checking uu_nl v0.0.25 (/Users/runner/work/coreutils/coreutils/src/uu/nl)
sccache: Compile terminated by signal 6
error: could not compile `coreutils` (test "tests")
Error: Process completed with exit code 1.
```

This affects at least #6112 and #6115.

```
    Checking uu_comm v0.0.26 (/Users/runner/work/coreutils/coreutils/src/uu/comm)
    Checking uu_uptime v0.0.26 (/Users/runner/work/coreutils/coreutils/src/uu/uptime)
sccache: Compile terminated by signal 6
error: could not compile `uu_numfmt` (lib test)
warning: build failed, waiting for other jobs to finish...
Error: Process completed with exit code 1.
```

--- a/.github/workflows/code-quality.yml
+++ b/.github/workflows/code-quality.yml
@@ -117,16 +117,21 @@ jobs:
           macos-latest) brew install coreutils ;; # needed for show-utils.sh
         esac
     - name: "`cargo clippy` lint testing"
-      shell: bash
-      run: |
-        ## `cargo clippy` lint testing
-        unset fault
-        CLIPPY_FLAGS="-W clippy::default_trait_access -W clippy::manual_string_new -W clippy::cognitive_complexity -W clippy::implicit_clone -W clippy::range-plus-one -W clippy::redundant-clone"
-        fault_type="${{ steps.vars.outputs.FAULT_TYPE }}"
-        fault_prefix=$(echo "$fault_type" | tr '[:lower:]' '[:upper:]')
-        # * convert any warnings to GHA UI annotations; ref: <https://help.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-a-warning-message>
-        S=$(cargo clippy --all-targets ${{ matrix.job.cargo-options }} ${{ steps.vars.outputs.CARGO_UTILITY_LIST_OPTIONS }} -- ${CLIPPY_FLAGS} -D warnings 2>&1) && printf "%s\n" "$S" || { printf "%s\n" "$S" ; printf "%s" "$S" | sed -E -n -e '/^error:/{' -e "N; s/^error:[[:space:]]+(.*)\\n[[:space:]]+-->[[:space:]]+(.*):([0-9]+):([0-9]+).*$/::${fault_type} file=\2,line=\3,col=\4::${fault_prefix}: \`cargo clippy\`: \1 (file:'\2', line:\3)/p;" -e '}' ; fault=true ; }
-        if [ -n "${{ steps.vars.outputs.FAIL_ON_FAULT }}" ] && [ -n "$fault" ]; then exit 1 ; fi
+      uses: nick-fields/retry@v2
+      with:
+        max_attempts: 3
+        retry_on: error
+        timeout_minutes: 90
+        shell: bash
+        command: |
+          ## `cargo clippy` lint testing
+          unset fault
+          CLIPPY_FLAGS="-W clippy::default_trait_access -W clippy::manual_string_new -W clippy::cognitive_complexity -W clippy::implicit_clone -W clippy::range-plus-one -W clippy::redundant-clone"
+          fault_type="${{ steps.vars.outputs.FAULT_TYPE }}"
+          fault_prefix=$(echo "$fault_type" | tr '[:lower:]' '[:upper:]')
+          # * convert any warnings to GHA UI annotations; ref: <https://help.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-a-warning-message>
+          S=$(cargo clippy --all-targets ${{ matrix.job.cargo-options }} ${{ steps.vars.outputs.CARGO_UTILITY_LIST_OPTIONS }} -- ${CLIPPY_FLAGS} -D warnings 2>&1) && printf "%s\n" "$S" || { printf "%s\n" "$S" ; printf "%s" "$S" | sed -E -n -e '/^error:/{' -e "N; s/^error:[[:space:]]+(.*)\\n[[:space:]]+-->[[:space:]]+(.*):([0-9]+):([0-9]+).*$/::${fault_type} file=\2,line=\3,col=\4::${fault_prefix}: \`cargo clippy\`: \1 (file:'\2', line:\3)/p;" -e '}' ; fault=true ; }
+          if [ -n "${{ steps.vars.outputs.FAIL_ON_FAULT }}" ] && [ -n "$fault" ]; then exit 1 ; fi
 
   style_spellcheck:
     name: Style/spelling

